### ISSUE TITLE
Castro should use particle format of Nyx/WarpX

### ISSUE THREAD

**Issue by zingale** [AMReX-Astro/Castro#95]
It seems that Castro outputs particles in its own ASCII format.  It should switch to use the same format as Nyx/WarpX and the other AMReX C++ codes.  This would allow yt to plot the particles.

Thanks to @atmyers for fixing up BoxLib yt particles: https://bitbucket.org/yt_analysis/yt/pull-requests/2497/boxlib-particle-support/diff#comment-30033180

**Comment by zingale** [comment]
The ASCII output is done in `CastroParticles.cpp` as:
```
TracerPC->WriteAsciiFile
```
which in turn is a member of the BoxLib `C_ParticleLib`.  It may be possible to simply change this to 
```
TracerPC->WritePlotFile
```
to make it yt happy.

**Comment by atmyers** [comment]
It looks to me like the above call to WriteAsciiFile only happens after restart, not every time a plot file is written out.

It looks like the full particle information *is* written in the NyX / WarpX format for the checkpoint files, but not for the plot files. We could also just add a call to ParticleCheckPoint in the writePlotFile routine in Castro_io.cpp, to put this information in the plot files as well.

**Comment by atmyers** [comment]
The following diff seems to produce plot files in the format yt can understand:

```
diff --git a/Source/Castro_io.cpp b/Source/Castro_io.cpp
index 6cb5187..f260c4c 100644
--- a/Source/Castro_io.cpp
+++ b/Source/Castro_io.cpp
@@ -807,6 +807,11 @@ Castro::writePlotFile (const std::string& dir,
                        ostream&       os,
                        VisMF::How     how)
 {
+
+#ifdef PARTICLES
+  ParticleCheckPoint(dir);
+#endif
+
     int i, n;
     //
     // The list of indices of State to write to plotfile.
```

**Comment by zingale** [comment]
good by me for you to do a PR on this.  No one is actively using particles, but we should have them work.

Also, this reminds me that the test suite doesn't know how to interpret particles either (separate issue).

**Comment by zingale** [comment]
was this addressed by #97?
can we close this?

**Comment by atmyers** [comment]
Yes - this is fixed now.
